financial-connections/src/main/java/com/stripe/android/financialconnections/ui/FinancialConnectionsSheetNativeActivity.kt

```diff
@@ -12,23 +14,49 @@ import androidx.navigation.NavHostController
 import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
+import com.airbnb.mvrx.MavericksView
 import com.airbnb.mvrx.viewModel
+import com.airbnb.mvrx.withState
 import com.stripe.android.financialconnections.navigation.NavigationDirections
+import com.stripe.android.financialconnections.presentation.CreateBrowserIntentForUrl
+import com.stripe.android.financialconnections.presentation.FinancialConnectionsSheetNativeViewEffect.OpenAuthFlowWithUrl
 import com.stripe.android.financialconnections.presentation.FinancialConnectionsSheetNativeViewModel
 
-internal class FinancialConnectionsSheetNativeActivity : AppCompatActivity() {
+internal class FinancialConnectionsSheetNativeActivity : AppCompatActivity(), MavericksView {
 
     val viewModel: FinancialConnectionsSheetNativeViewModel by viewModel()
 
+    private val startBrowserForResult = registerForActivityResult(StartActivityForResult()) {}
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        viewModel.onEach { postInvalidate() }
```

Comment:
is this necessary? shouldn't the state change triggers `invalidate` automatically?

Reply:
not for activities - https://github.com/airbnb/mavericks/issues/561#issuecomment-880870903

In our case, having some flow-controlled logic on the activity still makes sense, so decided to make it a mavericks view hence the workaround to get invalidate triggered.